mlflow/sagemaker/container/__init__.py

```diff
@@ -57,10 +86,39 @@ def _serve():
 
     Read the MLmodel config, initialize the Conda environment if needed and start python server.
     """
-    m = Model.load("/opt/ml/model/MLmodel")
-    if pyfunc.FLAVOR_NAME not in m.flavors:
-        raise Exception("Only supports pyfunc models and this is not one.")
-    conf = m.flavors[pyfunc.FLAVOR_NAME]
+    model_config_path = os.path.join(MODEL_PATH, "MLmodel")
+    m = Model.load(model_config_path)
+
+    if ENV_KEY_DEPLOYMENT_CONFIG in os.environ:
+        deployment_config = yaml.load(os.environ[ENV_KEY_DEPLOYMENT_CONFIG])
+        serving_flavor = deployment_config[DEPLOYMENT_CONFIG_KEY_FLAVOR_NAME]
+    else:
+        # Older models may not contain a deployment configuration
+        serving_flavor = get_serving_flavor(m)
+
+    if serving_flavor == mleap.FLAVOR_NAME:
+        # TODO(dbczumar): Host the scoring Java package on Maven Central so that we no
+        # longer require the container source for this flavor.
+        if _container_includes_mlflow_source():
+            _serve_mleap()
+        elif pyfunc.FLAVOR_NAME in m.flavors:
```

Comment:
I think we should fail if we want to run mleap but mleap is not available.

Reply:
I agree. Removed this segment and updated the exception.